test/unit/services/test_catalogue_item.py

```diff
@@ -2,2109 +2,1200 @@
-    )
-    # Mock `create` to return the created catalogue item
-    test_helpers.mock_create(catalogue_item_repository_mock, catalogue_item)
-
-    created_catalogue_item = catalogue_item_service.create(
-        CatalogueItemPostSchema(
-            catalogue_category_id=catalogue_item.catalogue_category_id,
-            manufacturer_id=catalogue_item.manufacturer_id,
-            **{
-                **CATALOGUE_ITEM_A_INFO,
-                "properties": [{"id": prop.id, "value": prop.value} for prop in catalogue_item.properties],
-            },
-        )
-    )
-
-    catalogue_category_repository_mock.get.assert_called_once_with(catalogue_item.catalogue_category_id)
-    catalogue_item_repository_mock.create.assert_called_once_with(
-        CatalogueItemIn(
-            catalogue_category_id=catalogue_item.catalogue_category_id,
-            manufacturer_id=catalogue_item.manufacturer_id,
-            **{
-                **FULL_CATALOGUE_ITEM_A_INFO,
-                "properties": properties,
-            },
-        )
-    )
-    assert created_catalogue_item == catalogue_item
-
-
-def test_create_with_non_existent_catalogue_category_id(
-    test_helpers, catalogue_category_repository_mock, catalogue_item_repository_mock, catalogue_item_service
-):
-    """
-    Test creating a catalogue item with a non-existent catalogue category ID.
-
-    Verify that the `create` method properly handles a catalogue item with a non-existent catalogue category ID, does
-    not find a catalogue category with such ID, and does not create the catalogue item.
-    """
-    catalogue_category_id = str(ObjectId())
-    properties = add_ids_to_properties(None, FULL_CATALOGUE_ITEM_A_INFO["properties"])
-
-    # Mock `get` to not return a catalogue category
-    test_helpers.mock_get(catalogue_category_repository_mock, None)
-
-    with pytest.raises(MissingRecordError) as exc:
-        catalogue_item_service.create(
-            CatalogueItemPostSchema(
-                catalogue_category_id=catalogue_category_id,
-                manufacturer_id=str(ObjectId()),
+        unit_repository_mock,
+        catalogue_item_service,
+        # Ensures all created and modified times are mocked throughout
+        # pylint: disable=unused-argument
+        model_mixins_datetime_now_mock,
+    ):
+        """Setup fixtures"""
+
+        self.mock_catalogue_item_repository = catalogue_item_repository_mock
+        self.mock_catalogue_category_repository = catalogue_category_repository_mock
+        self.mock_manufacturer_repository = manufacturer_repository_mock
+        self.mock_unit_repository = unit_repository_mock
+        self.catalogue_item_service = catalogue_item_service
+
+        with patch("inventory_management_system_api.services.catalogue_item.utils", wraps=utils) as wrapped_utils:
+            self.wrapped_utils = wrapped_utils
+            yield
+
+    def construct_properties_in_and_post_with_ids(
```

Comment:
could all the stuff we do with properties be abstracted even more? In a separate file maybe so it can be used by catalogue categories, catalogue items and items

Reply:
Yeah good idea, Viktor and I tried it briefly and didn't like how it was turning out at the time, so decided to leave it until items as we thought it would be clearer how we can simplify these then.